### PR TITLE
Show full page error on crash

### PR DESCRIPTION
**Pull request by faboweb**
> Thank you a lot for contributing to Cosmos Voyager!

### Issue

closes: #498

This passes any error in the main thread to the view and shows the full page error, showing the error message.
This will not shutdown the application on an error anymore.

❤️ Thank you!

### CODE REVIEW

File: app/src/renderer/main.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import Vuelidate from 'vuelidate'
 import shrinkStacktrace from '../helpers/shrink-stacktrace.js'
 import axios from 'axios'
 import Raven from 'raven-js'
-import {remote} from 'electron'
+import { remote } from 'electron'
```
Comment (jolesbi) [inline app/src/renderer/main.js]: i am in favour of these stylistic changes, but we should consider (agreeing on and) using a linter so no one changes this back / future styles are the same.

File: app/src/renderer/main.js [hunk 2]
```diff
@@ -51,32 +54,37 @@ async function main () {
...
 
-  const store = Store({ node })
+  store = Store({ node })
 
-  let connected = await store.dispatch('checkConnection')
-  if (connected) {
-    store.dispatch('nodeSubscribe')
-    store.dispatch('showInitialScreen')
-    store.dispatch('subscribeToBlocks')
+  let error = getQueryParameter('error')
```
Comment (jolesbi) [inline app/src/renderer/main.js]: i find it odd that we have to use url's even in a desktop app. anyone have any thoughts on this? not a big deal, just a curiosity.

File: app/src/renderer/main.js [hunk 3]
```diff
@@ -51,32 +54,37 @@ async function main () {
...
+  if (error) {
+    store.commit('setModalError', true)
+    store.commit('setModalErrorMessage', error)
+  } else {
+    let connected = await store.dispatch('checkConnection')
+    if (connected) {
+      store.dispatch('nodeSubscribe')
+      store.dispatch('showInitialScreen')
```
Comment (jolesbi) [inline app/src/renderer/main.js]: does this mean that if you get the full screen error, you have to sign back in 100% of the time?

File: app/src/renderer/vuex/modules/node.js [hunk 4]
```diff
@@ -43,13 +43,14 @@ export default function ({ node }) {
...
     },
-    nodeSubscribe ({commit, dispatch}) {
+    async nodeSubscribe ({commit, dispatch}) {
       if (state.stopConnecting) return
 
       // the rpc socket can be closed before we can even attach a listener
       // so we remember if the connection is open
       // we handle the reconnection here so we can attach all these listeners on reconnect
       if (!node.rpcOpen) {
+        await sleep(500)
```
Comment (jolesbi) [inline app/src/renderer/vuex/modules/node.js]: awesome

File: app/src/renderer/main.js (hunk 1 above)
Comment (faboweb) [reply app/src/renderer/main.js]: 👍 
vs code was messed up for a moment and reformatted stuff. Wrote in the chat about agreeing on one rule.

File: app/src/renderer/main.js (hunk 2 above)
Comment (faboweb) [reply app/src/renderer/main.js]: Currently this is the easiest solution. We can/should switch to a socket communication between the main thread and the browser.

File: app/src/renderer/main.js (hunk 3 above)
Comment (faboweb) [reply app/src/renderer/main.js]: The full page error will block you from doing anything but close the application. So you have to sign back in the next time.

File: app/src/renderer/vuex/modules/node.js (hunk 4 above)
Comment (faboweb) [reply app/src/renderer/vuex/modules/node.js]: This was needed so the process isn't blocked while continuously  trying to reconnect.

**Pull request merged**